Update adapter creation guide (#5421)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/core-versions.md

When you use dbt, you use a combination of `dbt-core` and an adapter plugin specific to your database. You can see the current list in [Supported Data Platforms](/docs/supported-data-platforms). Both `dbt-core` and dbt adapter plugins follow semantic versioning.

> `dbt-core` and adapter plugins coordinate new features and behind-the-scenes changes via the `dbt-adapters` interface. New features in adapters, and that `dbt-core` will leverage, are defined in `dbt-adapters`. These features are opt-in so they only impact adapters that explicitly implement them. This allows us to independently release adapters, `dbt-adapters`, and `dbt-core` without creating a broken experience for users.

Reviewer: this is great! i struggled to understand first time reading so tried to simplify
```suggestion
`dbt-core` and adapter plugins use the `dbt-adapters` interface to coordinate new features and behind-the-scenes changes. New adapter features are defined in `dbt-adapters` (which `dbt-core` will use). These features are opt-in, meaning they only impact adapters that explicitly implement them. This allows us to independently release adapters, `dbt-adapters`, and `dbt-core` without creating a broken experience for users.
```

Author: thank you! like the suggs. adjusted it a bit to remove anthropomorphism

> That means that (unlike core versions prior to 1.8) the minor and patch version numbers will likely differ between `dbt-core` and the adapter plugin(s) you have installed. 

Reviewer: ```suggestion
Unlike `dbt-core` versions before 1.8, the minor and patch version numbers might not match between `dbt-core` and the adapter plugin(s) you've installed. 
```

> For example, you may find you're using `dbt-core==1.8.0` with `dbt-snowflake==1.9.0`. Even though these do not have the same minor version, they can still work together as they both work with `dbt-adapters==1.8.0`. Patch releases can contain important bug or security fixes so it’s critical to stay up to date. 

Reviewer: keeping it consistent
```suggestion
For example, you might find you're using `dbt-core==1.8.0` with `dbt-snowflake==1.9.0`. Even though these don't have the same minor version, they can still work together as they both work with `dbt-adapters==1.8.0`. Patch releases can contain important bug or security fixes so it’s critical to stay up to date. 
```
